src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -3636,6 +3636,12 @@ private uint getJitFlags(ref CORJIT_FLAGS flags, uint sizeInBytes)
             flags.Set(CorJitFlag.CORJIT_FLAG_RELOC);
             flags.Set(CorJitFlag.CORJIT_FLAG_PREJIT);
             flags.Set(CorJitFlag.CORJIT_FLAG_USE_PINVOKE_HELPERS);
+#if X86
```

Comment:
Could you please use `compilation.TypeSystemContext.Target.Architecture == TargetArchitecture.X86` instead?